pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -144,32 +144,83 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, tc *v1alpha1.TektonChain
 	// Mark PreReconcile Complete
 	tc.Status.MarkPreReconcilerComplete()
 
-	// Check if a Tekton Chain InstallerSet already exists, if not then create one
-	labelSelector, err := common.LabelSelector(ls)
+	// Fetching and deleting the chains tektoninstallerset to delete `chains-config` configMap
+	// to handle the scenario when user upgrades i.e. in previous version `chains-config` configMap
+	// installerset was not there and with latest version we create separate installerset for
+	// `chains-config` configMap
+	chainlabelSelector, err := common.LabelSelector(ls)
 	if err != nil {
 		return err
 	}
-	existingInstallerSet, err := tektoninstallerset.CurrentInstallerSetName(ctx, r.operatorClientSet, labelSelector)
+
+	existingChainInstallerSet, err := tektoninstallerset.CurrentInstallerSetName(ctx, r.operatorClientSet, chainlabelSelector)
 	if err != nil {
 		return err
 	}
-	if existingInstallerSet == "" {
-		tc.Status.MarkInstallerSetNotAvailable("Chain InstallerSet not available")
 
-		createdIs, err := r.createInstallerSet(ctx, tc)
+	if existingChainInstallerSet != "" {
+		// If exists, then fetch the Tekton Chain InstallerSet
+		installedTIS, err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+			Get(ctx, existingChainInstallerSet, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			}
+		}
+
+		installerSetReleaseVersion := installedTIS.Labels[v1alpha1.ReleaseVersionKey]
+
+		if installerSetReleaseVersion != r.operatorVersion {
+			// Delete the existing Tekton Chain InstallerSet
+			err := r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+				Delete(ctx, existingChainInstallerSet, metav1.DeleteOptions{})
+			if err != nil {
+				logger.Error("failed to delete InstallerSet: %s", err)
+				return err
+			}
+
+			// Make sure the Tekton Chain InstallerSet is deleted
+			_, err = r.operatorClientSet.OperatorV1alpha1().TektonInstallerSets().
+				Get(ctx, existingChainInstallerSet, metav1.GetOptions{})
+			if err == nil {
+				tc.Status.MarkNotReady("Waiting for previous installer set to get deleted")
+				return v1alpha1.REQUEUE_EVENT_AFTER
+			}
+			if !apierrors.IsNotFound(err) {
+				logger.Error("failed to get InstallerSet: %s", err)
+				return err
+			}
+			return nil
+		}
+	}
+
+	// Check if a Tekton Chain Config InstallerSet already exists, if not then create one
+	configLabelSector, err := common.LabelSelector(configLs)
+	if err != nil {
+		return err
+	}
+
+	existingConfigInstallerSet, err := tektoninstallerset.CurrentInstallerSetName(ctx, r.operatorClientSet, configLabelSector)
+	if err != nil {
+		return err
+	}
+	if existingConfigInstallerSet == "" {
+		tc.Status.MarkInstallerSetNotAvailable("Chain Config InstallerSet not available")
+
+		createdIs, err := r.createConfigInstallerSet(ctx, tc)
 		if err != nil {
 			return err
 		}
 
 		return r.updateTektonChainStatus(tc, createdIs)
```

Comment:
reports chain config installerset is not available, immediately creating the new chain config installerset and updating the chain status.  
I do not know much here. Does this status will be helpful?

Reply:
Yeah this will be helpful as now we are creating separate installerset for chains-config configMap and once that is created then we proceed further to created the deployment using the different installerset